core/geometry/src/test/polyface/IndexedPolyfaceWalker.test.ts

```diff
@@ -0,0 +1,96 @@
+import { IndexedPolyface } from "../../polyface/Polyface";
+import { PolyfaceBuilder } from "../../polyface/PolyfaceBuilder";
+import { Checker } from "../Checker";
+
+function verifyMonotoneArraySearch(ck: Checker, data: number[], name: string) {
+  for (let i0 = 0; i0 + 1 < data.length; i0++) {
+    for (let k = data[i0]; k < data[i0 + 1]; k++) {
+      const i = IndexedPolyface.searchMonotoneNumbers(data, k);
+      if (ck.testDefined(i, "expect searchMonotoneNumbers to return defined"))
+        ck.testExactNumber(i0, i, `searchMonotoneNumbers(${k}) on ${name} indices returns the floor index`);
+    }
+  }
+}
+function verifyEdgeMates(ck: Checker, polyface: IndexedPolyface) {
+  const numFacets = polyface.facetCount;
+  let facetIndex;
+  for (facetIndex = 0; polyface.isValidFacetIndex(facetIndex); facetIndex++) {
+    const k0 = polyface.facetIndex0(facetIndex);
+    const k1 = polyface.facetIndex1(facetIndex);
+    for (let k = k0; k < k1; k++) {
+      const offsetWithinFacet = k - k0;
+      const walker = IndexedPolyfaceWalker.createAtEdgeIndex(polyface, k);
+      const walkerF = IndexedPolyfaceWalker.createAtFacetIndex(polyface, facetIndex, offsetWithinFacet);
+      ck.testTrue(walker.isSameEdge(walkerF), `createAtEdgeIndex(k) === createAtFacetIndex(f,k-k0) for f=${facetIndex}, k=${k}, k0=${k0}`);
+      const fNext = walker.nextAroundFacet();
+      const fNextPrev = fNext.previousAroundFacet();
+      checkWalkerPair(ck, "face steps", walker, fNext, false, fNextPrev);
+      const vNext = walker.nextAroundVertex();
+      const vNextPrev = vNext.previousAroundVertex();
+      checkWalkerPair(ck, "vertex steps", walker, vNext, true, vNextPrev);
+    }
+  }
+  ck.testExactNumber(numFacets, facetIndex, "confirm facet count");
+}
+function checkWalkerPair(ck: Checker, action: string, w: IndexedPolyfaceWalker, wNext: IndexedPolyfaceWalker, wNextMayBeUndefined: boolean, wNextPrev: IndexedPolyfaceWalker) {
+  if (ck.testTrue(w.isValid, "expect walker to be valid")) {
+    ck.testTrue(wNext.isValid || wNextMayBeUndefined, "expect next walker to be valid");
+    if (wNext.isValid) {
+      ck.testTrue(w.isDifferentEdgeInSamePolyface(wNext), `Expected ${action} action to move to a different edge`);
+      ck.testTrue(w.isSameEdge(wNextPrev), `Expected ${action} action to return to start`);
+    }
+  }
+}
+describe("IndexedPolyfaceWalker", () => {
+  it("edgeIndexToFacetIndex", () => {
+    const ck = new Checker();
+    const triIndices = [0, 3, 6, 9, 12, 15, 18];
+    verifyMonotoneArraySearch(ck, triIndices, "triangle");
+    const quadIndices = [0, 4, 8, 12, 16];
+    verifyMonotoneArraySearch(ck, quadIndices, "quad");
+    const mixedIndices = [0, 4, 12, 14, 20, 23, 30, 40, 43, 46, 49];
+    verifyMonotoneArraySearch(ck, mixedIndices, "mixed");
+    expect(ck.getNumErrors()).toBe(0);
+  });
+
+  it("edgeMates", () => {
+    const ck = new Checker();
+    //    5   2   4
```

Comment:
Is this the polyface visual? It's not clear to me. It would be great if you can visulize the polyface same as you did in `IndexedPolyfaceWalker.ts`
```
  # --------- # --------- #
  |   < < < B | F         |
  |         ^ | v         |
  |         ^ | v         |
  |         ^ | v         |
  | C > > > A | D > > >   |
  # --------- # --------- #
  |   < < < E |           |
  |           |           |
  |           |           |
  |           |           |
  # --------- # --------- #
  ```

Reply:
Yes, it's visual. But not the same. These are vertex indices, not edge indices. ASCII graphics would be overkill. To be addressed in next commit.